pod/MultiEffect/PodMultiEffect.cpp

```diff
@@ -0,0 +1,147 @@
+//#PodMultiEffect
+// ~~
+// Multiple effects on audio in.
+// Turn encoder to rotate through modes. (Only one effect active at a time)
+// Led brightness indicates knob position.
+//   * Note: when you cycle through modes it changes their settings!!
+// ~~
+// ## Mode one (red) : reverb
+//       *Knob one sets dry wet
+//       *Knob two sets reverb time (feedback)
+// ~~
+// ## Mode two (green) : delay
+//      *Knob one sets delay time
+//      *Knob two sets feedback level
+// ~~
+// ## Mode three (purple) : bitcrush 
+//      *Knob one sets lowpass filter cutoff
+//      *Knob two sets crush rate
+
+#include "daisysp.h"
+#include "daisy_pod.h"
+
+// Set max delay time to 0.75 of samplerate.
+#define MAX_DELAY static_cast<size_t>(48000 * 0.5f)
+
+using namespace daisysp;
+using namespace daisy;
+
+static DaisyPod  pod;
+
+static ReverbSc  rev;
+static DelayLine <float, MAX_DELAY>  del;
+static Bitcrush  crush;
+static Tone tone;
+static Parameter zero_one_l, zero_one_r,  deltime, cutoff, crushrate;
+
+int mode;
+
+static void AudioCallback(float *in, float *out, size_t size)
+{
+  float sigl, sigr, inl, inr, k1, k2, drywet, feedback, c;
+    
+    pod.UpdateAnalogControls();
+    pod.DebounceControls();
+    
+    mode += pod.encoder.Increment();
+    mode = (mode % 3 + 3) % 3;
+
+    k1 = zero_one_l.Process();
+    k2 = zero_one_r.Process(); 
+    
+    switch (mode)
+    {
+        case 0:
+   	    drywet = k1;
+	    rev.SetFeedback(k2);
+	    pod.led1.Set(k1,0,0);
+	    pod.led2.Set(k2,0,0);
+	    break;
+        case 1:
+ 	    del.SetDelay(deltime.Process());
```

Comment:
To avoid the quantization error artifacts, this should be called at the sample rate. I can't remember if the Parameter class has built in filtering or not, but since the Pod ADC stuff sets the knobs to the callback rate you can always use `daisysp::fonepole()` or the `Port` module to smooth the incoming control at audio rate.